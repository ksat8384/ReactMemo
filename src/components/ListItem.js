import React from 'react';
import {Text, StyleSheet, View} from 'react-native';

/**
 * Memoized list item component to avoid re-render unless its prop changes.
 *
 * This avoids re-rendering all the List items whenever the user adds a new item, insteads re-renders only the new list item
 */
const ListItem = React.memo(({item}) => {
  console.log('Render : ListItem');
  return (
    <View style={styles.item}>
      <Text style={styles.title}>{item.name}</Text>
    </View>
  );
});

export default ListItem;

const styles = StyleSheet.create({
  item: {
    backgroundColor: '#00ffff',
    padding: 20,
    marginVertical: 8,
  },
  title: {
    fontSize: 32,
  },
});
