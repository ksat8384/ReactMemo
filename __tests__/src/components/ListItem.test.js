import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import ListItem from "../../../src/components/ListItem";

describe("ListItem Component", () => {
  let component;
  const mockItem = { id: "a", name: "Batman" };

  beforeEach(() => {
    component = render(<ListItem item={mockItem} />);
  });

  afterEach(cleanup);

  it("renders correctly", () => {
    const { getByText, toJSON } = component;
    expect(toJSON()).toMatchSnapshot();
    expect(getByText("Batman")).toBeTruthy();
  });
});
