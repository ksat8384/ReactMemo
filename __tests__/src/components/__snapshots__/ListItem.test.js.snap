// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00ffff",
      "marginVertical": 8,
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
      }
    }
  >
    Batman
  </Text>
</View>
`;
